src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Collections/ArrayModule.fs

```diff
@@ -145,6 +145,53 @@ type ArrayModule() =
         ()
 
     [<Test>]
+    member this.ChunksOf() =
+
+        // int Seq
+        Assert.AreEqual([| [|1..4|]; [|5..8|] |], Array.chunksOf 4 [|1..8|])
+        Assert.AreEqual([| [|1..4|]; [|5..8|]; [|9..10|] |], Array.chunksOf 4 [|1..10|])
+
+        // string Seq
+        Assert.AreEqual([| [|"a"; "b"|]; [|"c";"d"|]; [|"e"|] |], Array.chunksOf 2 [|"a";"b";"c";"d";"e"|])
+
+        // empty Seq
+        Assert.AreEqual([||], Array.chunksOf 3 [||])
+
+        // null Seq
+        let nullArr:_[] = null
+        CheckThrowsArgumentNullException (fun () -> Array.chunksOf 3 nullArr |> ignore)
+
+        // invalidArg
+        CheckThrowsArgumentException (fun () -> Array.chunksOf 0 [|1..10|] |> ignore)
+        CheckThrowsArgumentException (fun () -> Array.chunksOf -1 [|1..10|] |> ignore)
+
+        ()
+
+    [<Test>]
+    member this.DivideInto() =
+
+        // int array
+        Assert.AreEqual([| [|1..4|]; [|5..8|]; [|9..10|] |], Array.divideInto 3 [|1..10|])
+        Assert.AreEqual([| [|1..4|]; [|5..8|]; [|9..11|] |], Array.divideInto 3 [|1..11|])
+        Assert.AreEqual([| [|1..4|]; [|5..8|]; [|9..12|] |], Array.divideInto 3 [|1..12|])
+
+        // string array
+        Assert.AreEqual([| [|"a"; "b"|]; [|"c";"d"|]; [|"e"|] |], Array.divideInto 3 [|"a";"b";"c";"d";"e"|])
```

Comment:
Some more cases where result is shorter than requested would be good, e.g. `divideInto 20 [| 1..10 |]` `divideInto 8 [| 1..9 |]` ?